lib/schemes/oauth2.js

```diff
@@ -108,6 +108,29 @@ export default class Oauth2Scheme {
       // Recommended by Auth0 on https://auth0.com/docs/api-auth/tutorials/nonce
       opts.nonce = nonce || nanoid()
     }
+    if (opts.code_challenge_method) {
+      switch (opts.code_challenge_method) {
+        case 'plain':
+        case 'S256': {
+          const state = this.generateRandomString()
+          localStorage.setItem('pkce_state', state)
+          const codeVerifier = this.generateRandomString()
```

Comment:
It cannot (AFAICT) - nano id is generated with a specific length (21 characters) - which is fine; but the minimum length I've seen (on 2 separate server implementations) is 28. Additionally, the length appears to require a certain modulus length (which I knew of back in Dec. - but its been awhile, and I've slept since then)

Reply:
Ok